Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp

```diff
@@ -117,10 +119,19 @@ void DNSResolveQueueCFNet::performDNSLookup(const String& hostname, Ref<Completi
     nw_parameters_set_context(parameters.get(), context.get());
     RetainPtr resolver = adoptCF(nw_resolver_create_with_endpoint(hostEndpoint.get(), parameters.get()));
 
-    nw_resolver_set_update_handler(resolver.get(), dispatch_get_main_queue(), makeBlockPtr([resolver, completionHandler = WTFMove(completionHandler)] (nw_resolver_status_t status, nw_array_t resolvedEndpoints) mutable {
-        if (status == nw_resolver_status_in_progress)
+    RELEASE_ASSERT_WITH_MESSAGE(isMainThread(), "Always create timer on the main thread.");
+    auto timeoutTimer = makeUnique<Timer>([resolver, completionHandler]() mutable {
+        nw_resolver_cancel(resolver.get());
+        completionHandler->complete(makeUnexpected(DNSError::Cancelled));
+    });
+    timeoutTimer->startOneShot(Seconds(timeoutForDNSResolution));
```

Comment:
You don't need `Seconds()` here, `timeoutForDNSResolution` already has the right type.